aspnetcore-webapi/RateLimitingDemo/RateLimitingDemo.UsingCustomMiddleware/Repositories/IProductCatalogRepository.cs

```diff
@@ -0,0 +1,13 @@
+﻿using RateLimitingDemo.UsingCustomMiddleware.Model;
+
+namespace RateLimitingDemo.UsingCustomMiddleware.Repositories;
+
+public interface IProductCatalogRepository
+{
+    void Create(Product product);
+    Product GetById(Guid id);
+    List<Product> GetAll();
+    void Update(Product product);
+    void Delete(Guid id);
+}
```

Comment:
Couldn't you create a class library for all the model classes and other stuff that's reused in both projects? That would be a much better approach than repeating the same code two times.

Reply:
done